cf-monitord/history.c

```diff
@@ -74,32 +74,34 @@ static void PutRecordForTime(CF_DB *db, time_t time, const Averages *values)
     WriteDB(db, timekey, values, sizeof(Averages));
 }
 
-static void Nova_SaveFilePosition(char *name, long fileptr)
+static void Nova_SaveFilePosition(const char *handle, char *name, long fileptr)
 {
     CF_DB *dbp;
+    char *key; key = StringConcatenate(2, handle, name);
```

Comment:
`StringConcatenate` allocates a new string. You don't free it anywhere.

Reply:
ACK

Reply:
```
char *key; key = StringConcatenate(2, handle, name);
```

Just  do:

```
char *key = StringConcatenate(2, handle, name);
```

Reply:
Could you please merge the declaration and initialization as Ole suggested above?

Reply:
ACK, pushed change.